client/src/types/ChatRoom.ts

```diff
@@ -0,0 +1,17 @@
+import { Message } from "./Message";
+import { User } from "./User";
+
+export type ChatRoomMin = {
+  id: string;
+  name: string;
+  type: "dm" | "channel";
+};
+
+export type ChatRoomMax = {
```

Comment:
What are these names? min/max?

Reply:
Because I want different attributes in Chatrooms. Like for sidebar list, I only need room id ,type and name. While loading chatroom messages, I need chatroom id, name, type, roomImage, members and messages.
Min represent minimum attributes
Max represent all attributes